src/inventory_ui.cpp

```diff
@@ -734,31 +836,79 @@ void inventory_selector::prepare_layout()
     layout_is_valid = true;
 }
 
-void inventory_selector::draw_inv_weight_vol( WINDOW *w, int weight_carried, units::volume vol_carried,
-        units::volume vol_capacity) const
+int inventory_selector::get_header_height() const
 {
-    int weight_capacity = u.weight_capacity();
-
-    mvwprintw( w, 0, 32, _( "Weight (%s): " ), weight_units() );
-    nc_color weight_color = weight_carried > weight_capacity ? c_red : c_ltgray;
-    wprintz( w, weight_color, "%6.1f", round_up( convert_weight( weight_carried  ), 1 ) );
-    wprintz( w, c_ltgray,   "/%-6.1f", round_up( convert_weight( weight_capacity ), 1 ) );
+    return display_stats || !hint.empty() ? 2 : 1;
+}
 
-    nc_color vol_color = vol_carried > vol_capacity ? c_red : c_ltgray;
-    mvwprintw( w, 0, 61, _( "Volume (L): ") );
-    wprintz( w, vol_color,  "%6.1f", round_up( to_liter( vol_carried  ), 1 ) );
-    wprintz( w, c_ltgray, "/%-6.1f", round_up( to_liter( vol_capacity ), 1 ) );
+int inventory_selector::get_column_height() const
+{
+    return getmaxy( w_inv ) - get_header_height() - 3;
 }
 
-void inventory_selector::draw_inv_weight_vol( WINDOW *w ) const
+void inventory_selector::draw_header( WINDOW *w ) const
 {
-    draw_inv_weight_vol( w, u.weight_carried(), u.volume_carried(), u.volume_capacity() );
+    trim_and_print( w, 0, screen_border_gap, getmaxx( w ) - screen_border_gap, c_ltgray, "%s", title.c_str() );
+    trim_and_print( w, 1, screen_border_gap, getmaxx( w ) - screen_border_gap, c_dkgray, "%s", hint.c_str() );
+
+    mvwhline( w, get_header_height(), 0, LINE_OXOX, getmaxx( w ) );
+
+    if( !display_stats ) {
+        return;
+    }
+
+    const player &dummy = get_player_for_stats();
+
+    static const int stats_count = 2;
+    static const int cells_count = 4;
+
+    using stat = std::array<std::string, cells_count>;
+    const auto disp = []( const std::string &caption, int cur_value, int max_value,
```

Comment:
Can you document this lambda explaining what it does. Doesn't need to be expansive.